File: ws/src/main/java/com/ahmeteminsaglik/ws/model/DataInfoDTO.java
package com.ahmeteminsaglik.ws.model;

public class DataInfoDTO<T> {
    private T value;
    private int index;
    private String explanation;
    private int totalSameNum = 0;

    public T getValue() {
        return value;
    }

    public void setValue(T value) {
        this.value = value;
    }

    public int getIndex() {
        return index;
    }

    public void setIndex(int index) {
        this.index = index;
    }

    public String getExplanation() {
        return explanation;
    }

    public void setExplanation(String explanation) {
        this.explanation = explanation;
    }

    public int getTotalSameNum() {
        return totalSameNum;
    }

    public void setTotalSameNum(int totalSameNum) {
        this.totalSameNum = totalSameNum;
    }

    @Override
    public String toString() {
        return "DataInfoDTO{" +
                "value=" + value +
                ", index=" + index +
                ", explanation='" + explanation + '\'' +
                ", totalSameNum=" + totalSameNum +
                '}';
    }
}

File: flutter_web_searchnode/lib/model/data_info_dto.dart
class DataInfoDTO {
  late String value;
  late int index;
  late String explanation;
  late int totalSameNum;

  DataInfoDTO(
      {required String value,
      required int index,
      required String explanation,
      required int totalSameNum}) {
    this.value = value;
    this.index = index;
    this.explanation = explanation;
    this.totalSameNum = totalSameNum;
  }

  DataInfoDTO.fromJson(Map<String, dynamic> json) {
    value = json['value'];
    index = json['index'];
    explanation = json['explanation'];
    totalSameNum = json['totalSameNum'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['value'] = this.value;
    data['index'] = this.index;
    data['explanation'] = this.explanation;
    data['totalSameNum'] = this.totalSameNum;
    return data;
  }

  @override
  String toString() {
    return 'DataInfoDto{value: $value, index: $index, explanation: $explanation, totalSameNum: $totalSameNum}';
  }
}
